### ISSUE TITLE
using new tensorflow

### ISSUE THREAD

**Issue by ARSadri** [Ivanlh20/tk_r_em#4]
This is both an issue and a simple solution for keras 3

True that you have included the environment, but given I want to test your code in my own setting I assumed I would pip3 install tensorflow, here is the necessary modification...notice that predict takes 6 args now, you can have **kwargs if you ask me because that is keras not your code.

so here is the change in the tk_r_em.py

    class Model(tf.keras.Model):
        def __init__(self, model_path):
          super(Model, self).__init__()
          # self.base_model = tf.keras.models.load_model(model_path, compile=False)
          
          model = tf.keras.layers.TFSMLayer(model_path, call_endpoint='serving_default')
          self.base_model = tf.keras.Sequential([model])
  
          self.base_model.compile()
          
      def call(self, inputs, training=None, mask=None):
          return self.base_model(inputs, training=training, mask=mask)
          
      def summary(self):
          return self.base_model.summary()
      
      def predict(self, x, batch_size=16, **kwargs):
          x_i_sh = x.shape
  
          # Expanding dimensions based on the input shape
          x = expand_dimensions(x)
  
          # Converting to float32 if necessary
          x = x.astype(np.float32)
  
          x_i_sh_e = x.shape
  
          # Adding extra row or column if necessary
          x = add_extra_row_or_column(x)
  
          batch_size = min(batch_size, x.shape[0])
  
          # Model prediction
          x = self.base_model.predict(x, batch_size, **kwargs)['gen_r_em_out']
  
          # Removing extra row or column if added
          x = remove_extra_row_or_column(x, x_i_sh_e)
  
          # Adjusting output dimensions to match input dimensions
          return adjust_output_dimensions(x, x_i_sh)

If you wish to support keras 3, you need to train using it otherwise legacy should be loaded as above.

**Comment by Ivanlh20** [comment]
Thanks for raising this.

This issue applied to the earlier TensorFlow/Keras-based implementation. Since the project has now been migrated to **ONNX Runtime**, `tk_r_em` no longer depends on TensorFlow or Keras for inference. The current README states that the models are shipped as lightweight **ONNX files** and run with **no TensorFlow dependency**, and the package metadata for version **2.0.0** lists `onnxruntime`, `onnxruntime-gpu`, and `onnxruntime-directml` as the relevant runtime options. ([[GitHub](https://github.com/Ivanlh20/tk_r_em/blob/main/README.md)][1])

So the Keras 3 compatibility workaround you suggested is no longer needed for the current codebase. That said, the report was useful and relevant for the older TensorFlow version of the project.

I’ll close this issue as resolved by the ONNX migration. Please open a new issue if you run into any problems with the current ONNX-based backend.

[1]: https://github.com/Ivanlh20/tk_r_em/blob/main/README.md "tk_r_em/README.md at main · Ivanlh20/tk_r_em · GitHub"
